src/transformers/modeling_utils.py

```diff
@@ -1277,17 +1277,32 @@ def _check_and_enable_flash_attn_2(
             )
 
         if not is_flash_attn_2_available():
-            raise ImportError(
-                "Flash Attention 2 is not available. Please refer to the documentation of https://github.com/Dao-AILab/flash-attention for"
-                " installing it. Make sure to have at least the version 2.1.0"
-            )
-        else:
             flash_attention_version = version.parse(importlib.metadata.version("flash_attn"))
-            is_flash_greater_than_2 = flash_attention_version >= version.parse("2.1.0")
-            if not is_flash_greater_than_2:
-                raise ValueError(
-                    f"You need flash_attn package version to be greater or equal than 2.1. Make sure to have that version installed - detected version {flash_attention_version}"
-                )
+            if torch.version.cuda:
+                install_message = "Please refer to the documentation of https://github.com/Dao-AILab/flash-attention to install it. Make sure to have at least the version 2.1.0."
+
+                if importlib.util.find_spec("flash_attn") is None:
+                    raise ImportError(f"The packaging flash_attn seem to be not installed. {install_message}")
```

Comment:
we already protect this block with `is_flash_attn_2_available()` why this is needed as well?

Reply:
This block of code is run only if `not is_flash_attn_2_available()`. This addition slightly precises the error messages shown.